plots/generic.py

```diff
@@ -0,0 +1,539 @@
+    :param filter_function: function
+        The function gets ONE neo.SpikeTrain as argument and if the return is
+        Truethy the spiketrain is included and if the return is Falsely it is
+        exluded.
+    :param histscale: float (default .1)
+        Portion of the figure used for the histograms on the right and upper
+        side.
+    :param labelkey: 0 | 1 | '0+1' (default) | 'annotation key' | None
+        * 0, 1: Set label according to first or second key in key_list.
+                Note that the first key is by default the list identification
+                key ('') when list of lists of spiketrains are given.
+        * '0+1': Two level labeling of 0 and 1
+        * annotation-key: Labeling each spiketrain with its value for given key
+        * None: No labeling
+        Note that only groups (-> see groupingdepth) can be labeled as bulks.
+        Alternatively you can color for an annotation key and show a legend.
+    :param markerargs: dict
+        Arguments dictionary is passed on to matplotlib.pyplot.plot()
+    :param separatorargs: dict | [dict, dict] | None
+        If only one dict is given and groupingdepth == 2 the arguments are
+        applied to the separator of both level. Otherwise the arguments are
+        of separatorargs[0] are applied to the level 1 and [1] to level 2.
+        Arguments dictionary is passed on to matplotlib.pyplot.plot()
+        To turn of separators set it to None.
+    :param legend: boolean
+        Show legend?
+    :param legendargs: dict
+        Arguments dictionary is passed on to matplotlib.pyplot.legend()
+    :param ax: matplotlib axis | None (default)
+        The axis onto which to plot. If None a new figure is created.
+        When an axis is given, the function can't handle the figure settings.
+        Therefore it is recommended to call seaborn.set() with your preferred
+        settings before creating your matplotlib figure in order to control
+        your plotting layout.
+    :param style: str
+        seaborn style setting. Default: 'ticks'
+    :param palette: string | sequence
+        Define the color palette either by its name or use a custom palette in
+        a sequence of the form ([r,g,b],[r,g,b],...).
+    :param colorcodes: str
+        seaborn colorcodes setting. By default this is set to 'colorblind' so
+        that any additional drawing into the plot using colorcodes ('r', 'g', ...)
+        are coherent with the default color palette ('Set2') of the plot.
+    :param context: 'paper'(default) | 'talk' | 'poster'
+        seaborn context setting which controls the scaling of labels. For the
+        three options the parameters are scaled by .8, 1.3, and 1.6 respectively.
+    :return: ax, axhistx, axhisty   <matplotlib axis handle>
+        * ax is handle of the dot display plot
+        * axhistx is handle of the histogram plot above the the dot display
+        * axhisty is handle of the histogram plot on the right hand side
+    """
```

Comment:
I think here it would be very necessary to put some examples, maybe a very simple one, one where you show how to separate rasters on the basis of list nesting, and one where you alter the appearance of the plot.
In etwa so:

 Example:
    # Make normal raster
    >>> st1 = neo.Spiketrain(np.random(...)...)
    >>> rasterplot(st1)...